hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,46 @@
+if answers['katello']
+  MONGO_DB_NAME = param('katello', 'pulp_db_name')
+else
+  MONGO_DB_NAME = param('foreman_proxy_content', 'pulp_db_name')
+end
+
+def migration
+  export_dir = '/tmp/engine_upgrade'
+  mongo_dir = '/var/lib/mongodb'
+  hiera_file = '/etc/foreman-installer/custom-hiera.yaml'
+
+  # Create directories and dump MongoDB
+  logger.info 'Stopping services except MongoDB'
+  Kafo::Helpers.execute('katello-service stop --exclude rh-mongodb34-mongod')
+  Kafo::Helpers.execute("mkdir #{export_dir} ; chmod 0700 #{export_dir}")
+  Kafo::Helpers.execute("Dir.exist?(#{export_dir})")
+  logger.info "Starting mongodump to #{export_dir}"
+  Kafo::Helpers.execute("mongodump --host localhost --out #{export_dir}")
+
+  # Move datafiles out of MongoDB directory so it will start
+  logger.info 'Export done, stopping MongoDB to remove old datafiles'
+  Kafo::Helpers.execute('systemctl stop rh-mongodb34-mongod')
+  logger.info "Moving contents from #{mongo_dir} to /var/mongo_backup"
+  Kafo::Helpers.execute('mkdir /var/mongo_backup ; chmod 0700 /var/mongo_backup')
+  Kafo::Helpers.execute('Dir.exist?(/var/mongo_backup)')
+  Kafo::Helpers.execute("mv #{mongo_dir}/* /var/mongo_backup")
+
+  # Start MongoDB and import the dump
+  logger.info 'Turning on MongoDB with wiredTiger and starting import'
+  Kafo::Helpers.execute('katello-service start --only rh-mongodb34-mongod')
+  Kafo::Helpers.execute("mongod --storageEngine wiredTiger --dbpath #{mongo_dir}")
+  Kafo::Helpers.execute("mongorestore --host localhost --db #{MONGO_DB_NAME} --quiet --dir {export_dir}")
+
```

Comment:
A failure case to notify the user about the backup in /var/mongo_backup might be useful.

Reply:
+1